src/Microsoft.Identity.Client/Config/AbstractConfigurationBuilder.cs

```diff
@@ -0,0 +1,274 @@
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.TelemetryCore;
+
+namespace Microsoft.Identity.Client.Config
+{
+    /// <summary>
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public abstract class AbstractConfigurationBuilder<T>
+        where T : AbstractConfigurationBuilder<T>
+    {
+        internal AbstractConfigurationBuilder(ApplicationConfiguration configuration)
+        {
+            Config = configuration;
+        }
+
+        internal ApplicationConfiguration Config { get; }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="httpClientFactory"></param>
+        /// <returns></returns>
+        public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)
+        {
+            Config.HttpClientFactory = httpClientFactory;
+            return (T)this;
+        }
+
+        internal T WithHttpManager(IHttpManager httpManager)
+        {
+            Config.HttpManager = httpManager;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        public T WithLoggingCallback(LogCallback loggingCallback)
+        {
+            Config.LoggingCallback = loggingCallback;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        internal T WithTelemetryCallback(ITelemetryReceiver telemetryReceiver)
+        {
+            Config.TelemetryReceiver = telemetryReceiver;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="onlySendFailureTelemetryData"></param>
+        /// <returns></returns>
+        public T WithOnlySendFailureTelemetryData(bool onlySendFailureTelemetryData)
```

Comment:
Do we have this in MSAL now?

Reply:
This is actually on the ITelemetryReceiver interface which is now public for telemetry consumption.  And yes, we honor this flag (off of the telemetry receiver) in TelemetryManager.cs

---
In reply to: [241519939](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/723#discussion_r241519939) [](ancestors = 241519939)